soci/gzip_indexer.go

```diff
@@ -0,0 +1,147 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package soci
+
+// #cgo CFLAGS: -I${SRCDIR}/../c/
```

Comment:
If we decide to go in the direction of defining SOCI as a module, then yes. Right now `soci` is just a package within soci-snapshotter project.

Reply:
It doesn't. It still uses `libindexer` functions. If we really want to make it available as a package others can use, we need to remove all `libindexer` code and rewrite it in Go. I don't know if we want that as part of this PR. OTOH, since we are creating a Go interface for the indexer, it should be part of this PR.

Reply:
This PR's intention is to provide an easily usable wrapper, so nobody deals with calling C code throughout the snapshotter. I will create an issue to track rewriting efforts and this will happen in another code change.